### PR TITLE
Tape Recorder Housekeeping

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request
ports
https://github.com/tgstation/tgstation/pull/51653
https://github.com/tgstation/tgstation/pull/55881
https://github.com/tgstation/tgstation/pull/55908
from tgstation. adds wires to recorders

## Why It's Good For The Game

Tape recorders r cool.

## Changelog

:cl: Erika - porting from /tg/ mostly.
add: Tape Recorders now have a radial menu when used
add: Tape recorders now make noises
add: Tape recorders now have wires (can be remotely used) 
/:cl:

### CODE REVIEW

**Comment by thgvr** [comment]
fixed ur pr body for u

File: code/game/objects/items/devices/taperecorder.dm [hunk 1]
```diff
@@ -38,18 +55,26 @@
...
 			return
-		mytape = I
-		to_chat(user, "<span class='notice'>You insert [I] into [src].</span>")
+		mytape = item
+		to_chat(user, span_notice("You insert [item] into [src]."))
+		playsound(src, 'sound/items/taperecorder/taperecorder_close.ogg', 50, FALSE)
 		update_appearance()
+	if(item.tool_behaviour == TOOL_SCREWDRIVER)
+		to_chat(usr, span_notice("You [open_panel ? "close" : "open"] [src]s panel."))
+		open_panel = !open_panel
```
Comment (FalloutFalcon) [inline code/game/objects/items/devices/taperecorder.dm]: this should use /screwdriver_act

**Comment by rye-rice** [comment]
update this pr in case its missing something before i merge it tommorow at 3am

**Comment by generalthrax** [comment]
Merge it tomorrow at 3am NOW

**Review by FalloutFalcon — DISMISSED** [review]
It could still use screwdriver act instead of having it in attackby! its kinda important so things can acctually trust COMSIG_ATOM_SCREWDRIVER_ACT.

**Comment by Erikafox** [comment]
> It could still use screwdriver act instead of having it in attackby! its kinda important so things can acctually trust COMSIG_ATOM_SCREWDRIVER_ACT.

donezo

**Review by Sun-Soaked — APPROVED** [review]
everything seems to be in order.

**Pull request merged**